### ISSUE TITLE
关于复现的问题

### ISSUE THREAD

**Issue by ZealBear** [vvhj/TSGCNeXt#9]
感谢您做出的杰出工作，我目前正在学习这个方向的知识，想对这篇论文的代码进行复现，但是可能是因为超参数的原因，复现结果不尽如人意，请问您方便提供一下模型训练时的超参数吗，谢谢！

**Comment by vvhj** [comment]
您方便提供一下用到的超参和问题吗？我帮您排查一下，训练参数尤其时batchsize，使用的显卡个数和ema的参数设置对结果的影响比较大。

**Comment by ZealBear** [comment]
您好，感谢您的帮助，我由于在单张3090Ti上进行的训练，所以没有使用论文中所提到的分布式训练的参数，ema参数是直接使用代码里的没有修改，batchsize由于显存不够所以没法设置成论文中所说的128，是这个的问题吗？

我的训练超参如下：

训练指令：python main_modern.py --config config/nturgbd-cross-sub/TSGCNextmodern_252.yaml --batch_size 96 --lr 4e-3 --update_freq 2 --model_ema true --model_ema_eval true

运行代码时显示的参数如下所示
Namespace(aa='rand-m9-mstd0.5-inc1', auto_resume=True, batch_size=96, clip_grad=None, color_jitter=0.4, config='config/nturgbd-cross-sub/TSGCNextmodern_252.yaml', crop_pct=None, cutmix=1.0, cutmix_minmax=None, data_set='IMNET', device='cuda', disable_eval=False, dist_eval=True, dist_on_itp=False, dist_url='env://', distributed=False, enable_wandb=False, epochs=300, eval=False, feeder='feeders.feeder_ntu.Feeder', finetune='', head_init_scale=1.0, imagenet_default_mean_and_std=True, layer_decay=1.0, layer_scale_init_value=1e-06, local_rank=-1, log_dir='./newwork_dir/ntu60/xsub252/TSGCNext3_jointmodern', lr=0.004, min_lr=1e-06, mixup=0.8, mixup_mode='batch', mixup_prob=1.0, mixup_switch_prob=0.5, model='model.tsmartaagcnext_3.Model', model_args={'num_class': 60, 'num_point': 25, 'num_person': 2, 'graph': 'graph.ntu_rgb_d.Graph', 'graph_args': {'labeling_mode': 'spatial'}, 'drop_out': 0.4}, model_ema=True, model_ema_decay=0.9999, model_ema_eval=True, model_ema_force_cpu=False, model_key='model|module', model_prefix='', momentum=0.9, nb_classes=1000, nesterov=True, nomixup_epochs=0, num_workers=10, opt='adamw', opt_betas=None, opt_eps=1e-08, output_dir='./newwork_dir/ntu60/xsub252/TSGCNext3_jointmodern', pin_mem=True, project='convnext', recount=1, remode='pixel', reprob=0.25, resplit=False, resume='', save_ckpt=True, save_ckpt_freq=1, save_ckpt_num=3, seed=0, smoothing=0.1, start_epoch=0, test_batch_size=32, test_feeder_args={'data_path': 'data/ntu/NTU60_CS.npz', 'split': 'test', 'window_size': 256, 'p_interval': [0.95], 'vel': False, 'bone': False, 'debug': False}, train_feeder_args={'data_path': 'data/ntu/NTU60_CS.npz', 'split': 'train', 'debug': False, 'random_choose': False, 'random_shift': False, 'random_move': False, 'window_size': 256, 'normalization': False, 'random_rot': True, 'p_interval': [0.5, 1], 'vel': False, 'bone': False}, train_interpolation='bicubic', update_freq=2, use_amp=False, wandb_ckpt=False, warmup_epochs=20, warmup_steps=-1, weight_decay=0.05, weight_decay_end=None, work_dir='./newwork_dir/ntu60/xsub252/TSGCNext3_jointmodern', world_size=1)

config中的内容没有修改

最后的最好的结果是test_acc1_ema: 89.97，test_acc1: 89.36

**Comment by vvhj** [comment]
可以调一下batch_size和update_freq，您现在实际的batch_size是96x2=192，可以把batch_size设置为64试试。

**Comment by ZealBear** [comment]
> 可以调一下batch_size和update_freq，您现在实际的batch_size是96x2=192，可以把batch_size设置为64试试。

好的谢谢您的回复，我去试一下

**Comment by vvhj** [comment]
ntu60上实际上有更高的精度，建议您调一下k的大小，在老版本里k=7效果会更好些

**Comment by ZealBear** [comment]
> ntu60上实际上有更高的精度，建议您调一下k的大小，在老版本里k=7效果会更好些

好的我多跑几次分别测试一下

**Comment by ZealBear** [comment]
您好,方便问一下您当时训练的时候python是什么版本吗,我觉得可能有包版本不一样的原因但我试了几个python的版本安装requirements都会有版本冲突的问题

**Comment by vvhj** [comment]
Python 3.8.12

**Comment by ZealBear** [comment]
用了这些超参数虽然还有一些偏差但是已经比之前好很多了，谢谢您的帮助
我还想请问一下HRnet中的NTU60_CS.npz是怎么生成的呢？还是说和ntu里的NTU60_CS.npz一样

**Comment by vvhj** [comment]
从stgcn++的数据集中转换的，近期更新数据集相关的文件，谷歌网盘满了（尴尬）

**Comment by ZealBear** [comment]
好的，麻烦您了，如果方便的话可否在百度网盘之类的国内网盘也放一份呢，我在谷歌网盘下载速度太慢了，谢谢
